### ISSUE TITLE
Inconsistent behaviour in parsing `select` arguments

### ISSUE THREAD

**Issue by stinodego** [pola-rs/polars#6606]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

I was investigating the type hints of `select`, and ran into this issue. The parsing of input is inconsistent between single input, list input, and keyword input.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({"a": [1, 2], "b": [3, 4]})

# String interpreted as column name - GOOD
result = df.select("a")

┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 1   │
│ 2   │
└─────┘

# String interpreted as literal - BAD?
result = df.select(q="a")

┌─────┐
│ q   │
│ --- │
│ str │
╞═════╡
│ a   │
└─────┘

# TypeError - not allowed - BAD?
result = df.select(1)

# Integer interpreted as literal - GOOD?
result = df.select([1])

┌─────────┐
│ literal │
│ ---     │
│ i32     │
╞═════════╡
│ 1       │
└─────────┘

# Unrelated: Mixed select is allowed - seems like we should error here - BAD
result = df.select("a", reverse=True)

┌─────┬─────────┐
│ a   ┆ reverse │
│ --- ┆ ---     │
│ i64 ┆ bool    │
╞═════╪═════════╡
│ 1   ┆ true    │
│ 2   ┆ true    │
└─────┴─────────┘
```

### Expected behavior

We should have consistent behaviour between passing input between all three argument types (regular, list, and keyword).

I would propose:
* Expressions are taken as-is
* Strings are interpreted as column names
* Other values are interpreted literals

As far as I can see, this needs to be fixed in the Python LazyFrame parsing logic in `select` / `selection_to_pyexpr_list`.

This may also affect other methods besides `select`.

I'd like to pick this up if we decide which way to go with this one.

### Installed versions

master branch

**Comment by alexander-beedie** [comment]
I don't see any problem with mixed select (there are calling patterns where it's helpful), but I definitely agree with getting `select` and `with_columns` to behave more consistently! 

FYI: as part of https://github.com/pola-rs/polars/pull/6615 I did rationalise a few lines of `with_columns` to call `selection_to_pyexpr_list` instead of copying the logic inline. Sounds like we need to do some more of that ;)

**Comment by stinodego** [comment]
> I don't see any problem with mixed select (there are calling patterns where it's helpful)

Fair enough, I wasn't quite sure about that one. There's indeed also something to be said for allowing it. I'll remove that from the examples.

**Comment by alexander-beedie** [comment]
Ahh; the str-to-col inconsistency can be easily fixed by setting `str_to_lit=False` when evaluating `named_exprs` (which is what `selection_to_pyexpr_list` sets). I've just updated https://github.com/pola-rs/polars/pull/6615 accordingly 👌

**Comment by alexander-beedie** [comment]
> Yes, I suppose that's an easy fix. But I wanted to double-check that this is not intentional.

It wasn't - I just missed the param...😓 
The intention was always to behave the same as non-kwarg exprs.

**Comment by stinodego** [comment]
Now the issue remains that there is a discrepancy between list-input and regular input.

**Comment by alexander-beedie** [comment]
> Now the issue remains that there is a discrepancy between list-input and regular input.

Queuing up a fix for that one too 👍
